### PR TITLE
Permissions users api tests

### PR DESCRIPTION
**Pull request by AndrzejPodo**
- adds and tests permissions handling on the users list
- integrates backend with front-end

### CODE REVIEW

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/AuthChecker.java [hunk 1]
```diff
@@ -23,15 +24,15 @@ public static Permission boardModifyPermission(Whiteboard board) {
     // check that user is authenticated
     UserService userService = UserServiceFactory.getUserService();
     if (userService.isUserLoggedIn()) {
-      User user =
-          ofy().load().type(User.class).filter("googleAccId", userService.getCurrentUser().getUserId()).first().now();
```
Comment (PaulKovalov) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/AuthChecker.java:27]: our linters are definitely fighting with each other XD

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/BoardServlet.java [hunk 2]
```diff
@@ -37,19 +37,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     if (boardIdParam != null) {
       long boardId = Long.parseLong(boardIdParam);
+      // check if user can access the board
       Whiteboard board = ofy().load().type(Whiteboard.class).id(boardId).now();
       if (board == null) {
         response.getWriter().println("Board with this id doesn't exist");
         response.sendError(BAD_REQUEST);
         return;
       }
-      // check if user can access the board
```
Comment (PaulKovalov) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/BoardServlet.java:46]: why this comment moved  above?

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java [hunk 3]
```diff
@@ -78,6 +92,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
+    Permission perm = AuthChecker.userListModifyPermission(role, board);
+    if (perm != Permission.GRANTED) {
```
Comment (PaulKovalov) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java:96]: It must be equalsTo? I guess in Java strings are compared using equals only

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java [hunk 4]
```diff
@@ -116,6 +143,12 @@ public void doDelete(HttpServletRequest request, HttpServletResponse response) t
       return;
     }
 
+    Permission perm = AuthChecker.userListModifyPermission(boardRole.role, boardRole.getBoard());
+    if (perm != Permission.GRANTED) {
```
Comment (PaulKovalov) [inline sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java:147]: same comment about comparing strings

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java (hunk 3 above)
Comment (AndrzejPodo) [reply sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java:96]: https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals

Comment (PaulKovalov) [reply sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/UserListServlet.java:96]: didn't know that, thank you

File: sticknotes-frontend/src/app/boards-list/boards-list.component.ts [hunk 5]
```diff
@@ -18,10 +18,13 @@ export class BoardsListComponent implements OnInit {
   constructor(private dialog: MatDialog, private boardApiService: BoardApiService, private userService: UserService) { }
 
   ngOnInit(): void {
-    this.boardApiService.myBoardsList().subscribe(boards => {
-      this.myBoards = boards;
+    //ensuring right order of fetching data
+    this.userService.getUser().subscribe(user => { 
+      this.user = user
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/boards-list/boards-list.component.ts:23]: How do we use field user here?

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/boards-list/boards-list.component.ts:23]: I think we can remove local user field

Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/boards-list/boards-list.component.ts:23]: It is now used in .html

File: sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/BoardServlet.java (hunk 2 above)
Comment (AndrzejPodo) [reply sticknotesbackend/src/main/java/com/google/sticknotesbackend/servlets/BoardServlet.java:46]: by accident, sorry.

File: sticknotes-frontend/src/app/boards-list/boards-list.component.ts (hunk 5 above)
Comment (PaulKovalov) [reply sticknotes-frontend/src/app/boards-list/boards-list.component.ts:23]: Yes, but in html we can check `*ngIf="myBoards"`

**Review by PaulKovalov — CHANGES_REQUESTED** [review]
Please remove user field from boards-list.component.ts

**Review by PaulKovalov — APPROVED** [review]

**Pull request merged**